### PR TITLE
add a pager

### LINKED ISSUE #12: [Feature Request] add a simple pager implementation
I know the role of this lib is not about the view, but I think it would beneficiate of a simple pager like:

``` php

    public function isCurrent($page)
    {
        return $this->getCurrentPage() === $page;
    }

    public function getPages()
    {
        return range($this->getSliceStart(), $this->getSliceEnd());
    }

    public function getNumberOfPages()
    {
        return ceil($this->totalCount / $this->limit) ?: 1;
    }

    private function getSliceStart()
    {
        return min(max(1, $this->getNumberOfPages() - 3), max(1, $this->getCurrentPage() - 3));
    }

    private function getSliceEnd()
    {
        return max(1, min($this->getNumberOfPages(), $this->getCurrentPage() + 3));
    }

```

Thoughts?

### PR DESCRIPTION
**Pull request by docteurklein**
closes #12

### CODE REVIEW

File: src/Porpaginas/Pager.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    private $limit;
+    private $currentPage;
+
+    public function __construct($totalCount, $limit, $currentPage)
+    {
+        $this->totalCount = $totalCount;
+        $this->limit = $limit;
+        $this->currentPage = max(1, $currentPage);
+    }
+
```
Comment (LaurentBrieu) [inline src/Porpaginas/Pager.php]: Plz remove that extra line ;)

**Comment by beberlei** [comment]
@docteurklein Can you add documentation to README.md and a small example how you would use this in Symfony controller for example?

**Comment by docteurklein** [comment]
@beberlei I added an example in the README.

**Comment by docteurklein** [comment]
The advantage is that it removes entirely the need for pagerfanta or knppaginator.

**Comment by docteurklein** [comment]
A few ideas come to my mind:

 - [] add a `Pager::fromPage` named constructor.
 - [] add a twig function to do this from the view
 - [] include a default template (?)

File: README.md [hunk 2]
```diff
@@ -157,3 +157,25 @@ We found a total of <strong>{{ porpaginas_total(users) }}</strong> users:
 ## Pager Library Support
 
 * For Pagerfanta use the ``Porpaginas\Pagerfanta\PorpaginasAdapter`` and pass it a result as first argument.
+
+## Embedded Pager
+
+You can use the `Porpaginas\Pager` class to help you get a slice of previous and next pages to display:
+
+``` php
+$pager = new Porpaginas\Pager($page->totalCount(), $page->getCurrentLimit(), $page->getCurrentPage());
```
Comment (beberlei) [inline README.md]: That is too complicated, can we have a static factory method?

Comment (docteurklein) [inline README.md]: yes, good idea

Comment (docteurklein) [inline README.md]: @beberlei done

**Comment by beberlei** [comment]
This is already a good first start, i merged it :-)

**Comment by docteurklein** [comment]
thanks!

**Pull request merged**